Source/WebCore/dom/AbortSignal.cpp

```diff
@@ -86,6 +103,25 @@ AbortSignal::AbortSignal(ScriptExecutionContext* context, Aborted aborted, JSC::
 
 AbortSignal::~AbortSignal() = default;
 
+void AbortSignal::addSourceSignal(AbortSignal& signal)
+{
+    if (signal.isDependent()) {
+        for (auto& sourceSignal : signal.sourceSignals())
+            addSourceSignal(sourceSignal);
+        return;
+    }
+    ASSERT(!signal.aborted());
+    ASSERT(!signal.isDependent());
```

Comment:
This ASSERT seems unnecessary, unless it's possible for a single to move from dependent to non-dependent; I don't see any asynchronous calls so it seems impossible for this to ever be `!isDependent()`

Reply:
I can drop them, the reason I added them is that they were in the spec text.

Reply:
If the specifications suggests asserting this precondition, I have no objection to leaving it in the code. It just seemed weird to branch on some value, then assert the value again.

Reply:
In the spec things are organized a bit differently such that this is not immediately obvious. Your code seems a bit more elegant due to the separation of "add source signals". Perhaps something to contribute upstream at some point.